index.html

```diff
@@ -0,0 +1,96 @@
+    <script type="text/javascript">
+        //<![CDATA[
+
+        function getPreviousRouteList() {
+            var street = getURLParameterByName("street");
+            if (street != null) {
+                document.getElementById("txtStreetName").value = street;
+                getRouteListByStreetName();
+            }
+        }
+
+        function getRouteListByStreetName() {
+            var txtStreetName = document.getElementById("txtStreetName").value.replace(/^\s+|\s+$/gm,'');
+            if (!txtStreetName || txtStreetName.trim() == "") {
+                alert("Street is required to get routes!");
+                return;
+            }
+
+            var oOptions = {
+                method: "POST",
+                postBody: "\n{\n\n\"params\": {\n\n\"stopName\": \"%" + txtStreetName + "%\"\n\n}\n\n}",
+				contentType: "application/json",
+                requestHeaders: [ "X-AppGlu-Environment", "staging", "Authorization", "Basic V0tENE43WU1BMXVpTThWOkR0ZFR0ek1MUWxBMGhrMkMxWWk1cEx5VklsQVE2OA==", "Accept", "application/json"],
+                onSuccess: function (oXHR, oJson) {
+                    displayRouteList(oXHR.responseText);
+                },
+                onFailure: function (oXHR, oJson) {
+                    alert("An error occurred requesting route: " + oXHR.statusText);
+                }
+            };
+
+            var oRequest = new Ajax.Request("https://api.appglu.com/v1/queries/findRoutesByStopName/run", oOptions);
+        }
+
+        function displayRouteList(response) {
+            var txtStreetNameToShow = document.getElementById("txtStreetName").value;
+            var txtStreetName       = replaceAll(document.getElementById("txtStreetName").value, " ", "+");
+            var divRouteList        = document.getElementById("divRouteList");
+            var routeLine           = "";
+        	var allRoutes           = response.evalJSON(true);
+
+        	if (allRoutes.rows.length == 0) {
+        		divRouteList.innerHTML = "<p><b>Street not found!</b></p>";
+        	}
+        	else {
+        		var htmlTableHeader  = "<p><b>" + allRoutes.rows.length + " route/routes found for street \"" + txtStreetNameToShow + "\"</b></p><table border=0><thead><th>Line #</th><th>Line Name</th></thead><tbody>";
+        		var htmlTableContent = "";
+        		allRoutes.rows.each(function(aRoute){
+                    routeLine = replaceAll(aRoute.longName, " ", "+");
+        			htmlTableContent = htmlTableContent + "</td><td>" + aRoute.shortName
+        												+ "</td><td><a href=routeDetails.html?routeId=" + aRoute.id + "&routeNo=" + aRoute.shortName + "&routeLine=" + routeLine + "&street=" + txtStreetName + ">" + aRoute.longName + "</a>"
+                                                        + "</td></tr>";
+				});
+				divRouteList.innerHTML = htmlTableHeader + htmlTableContent + "</tbody></table>";
+                localStorage.setItem('divRouteList', divRouteList.innerHTML);

```

Comment:
Sorry, I forgot to remove it. It was an attempt to store the last route
search instead of performing the same search again to display the routes
list.


2016-03-02 17:18 GMT-03:00 André Luiz Cardoso <notifications@github.com>:

> In index.html
> <https://github.com/otaviofontana/arcTouchOtavioFontana/pull/1#discussion_r54784269>
> :
>
> > +        	var allRoutes           = response.evalJSON(true);
> > +
> > +        	if (allRoutes.rows.length == 0) {
> > +        		divRouteList.innerHTML = "<p><b>Street not found!</b></p>";
> > +        	}
> > +        	else {
> > +        		var htmlTableHeader  = "<p><b>" + allRoutes.rows.length + " route/routes found for street \"" + txtStreetNameToShow + "\"</b></p><table border=0><thead><th>Line #</th><th>Line Name</th></thead><tbody>";
> > +        		var htmlTableContent = "";
> > +        		allRoutes.rows.each(function(aRoute){
> > +                    routeLine = replaceAll(aRoute.longName, " ", "+");
> > +        			htmlTableContent = htmlTableContent + "</td><td>" + aRoute.shortName
> > +        												+ "</td><td><a href=routeDetails.html?routeId=" + aRoute.id + "&routeNo=" + aRoute.shortName + "&routeLine=" + routeLine + "&street=" + txtStreetName + ">" + aRoute.longName + "</a>"
> > +                                                        + "</td></tr>";
> > +				});
> > +				divRouteList.innerHTML = htmlTableHeader + htmlTableContent + "</tbody></table>";
> > +                localStorage.setItem('divRouteList', divRouteList.innerHTML);
>
> Why are you saving the html into localStorage?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/otaviofontana/arcTouchOtavioFontana/pull/1/files#r54784269>
> .
>